src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/NameMatcher.cs

```diff
@@ -25,11 +25,27 @@ private NameMatcher(string typeName, string? memberName = null)
 
         public static NameMatcher MatchType(string typeName) => new NameMatcher(typeName);
 
-        public static NameMatcher MatchPropertyAccess(string typeName, string memberName) => new(typeName, memberName);
+        public static NameMatcher MatchMemberAccess(string memberName)
+        {
+            if (memberName is null)
+            {
+                throw new ArgumentNullException(nameof(memberName));
+            }
+
+            var idx = memberName.LastIndexOf('.');
+            if (idx < 0)
+            {
+                throw new ArgumentException("Member name must be fully qualified");
+            }
+
+            return new NameMatcher(memberName.Substring(0, idx), memberName.Substring(idx + 1));
```

Comment:
We'd still have to allocate strings to pass to NameMatcher. We could look into making NameMatcher work with ROS instead of strings, but that would be a larger change that we should probably track separately.